### DISCUSSION TITLE
Eager loading of record in params

### DISCUSSION THREAD

**Discussion by Itachi-weasel** [excid3/noticed#147] [Q&A]
Is there any way to eager load the record in params? It's generating N+1 queries when I display the last 10 unread notifications.

eg: `current_user.notifications.unread.order(id: :desc).limit(10)`
N+1 queries are generated for each post i.e stored in the params

**Comment by excid3** [comment]
To eager load, pass in the ID of the associated record(s) to `params` instead of the object. Then you can pluck the IDs and load them in a batch.

The N+1 is not a problem as long as you fragment cache when you render the notification.

**Reply by Itachi-weasel** [reply]
Thanks for the tip. In my case, fragment caching and proper pagination will be more beneficial.
